### DISCUSSION TITLE
dataconnection vs datacontext

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#3142] [General]
When I use Linq2SQL, I use datacontext to ddo all the dirty work.
Ssuprisingly, I found a dataconnection class here that can do all the work of datacontext?
SSo what is the deal here? Am I supposed to get rid of datacontext and use dataconnection directly?
What is the difference? performance wise? heavy/light? Why do we have classes here? Since we have dataconnection, why do we still need datacontext?

**Comment by MaceWindu** [comment]
`DataConnection`  implements persistent connection to database where it holds open database connection from moment it opened till instance disposal or explicit `Close` call.

`DataContext` is just a wrapper over `DataConnection` with (by default) open-execute-close semantic, where connection is acquired and released for each command.

I personally don't see a lot of value in `DataContext`. Also some of existing APIs lack `DataContext` support and require `DataConnection` instance (we fix it occasionally, but still some of such APIs exist).
